codalab/worker/file_util.py

```diff
@@ -171,11 +184,192 @@ def do_unzip(filename):
         do_unzip(fileobj_or_name)
 
 
-def open_file(file_path, mode='r', compression_type=CompressionTypes.UNCOMPRESSED):
+def open_indexed_tar_gz_file(path):
```

Comment:
This seems like a very specific structure...should we build an abstraction around this instead of returning a tuple (either way, should type annotate).  We might consider moving this logic into a separate file so that `file_util.py` doesn't become too cluttered.

Reply:
I found a better way to do this -- the tuple was initially created to work around a bug where the file kept getting closed. Now, I use a context manager that just returns the SQLiteIndexedTar file object itself.